# Question Four
##### ABC Investments

In [17]:
using JuMP, Gurobi

# all in millions
minInvestment = [3 2 9 5 12 4]
maxInvestment = [27 12 35 15 46 18]

# percent return
expectedRtn = [0.13 0.09 0.17 0.1 0.22 0.12]

# total amount = 80 million
totalInvestment = 80

n = length(expectedRtn)

m = Model(solver = GurobiSolver(OutputFlag = false))

# binary selections
@variable(m, x[1:n], Bin)
@variable(m, y[1:n] >= 0)

@constraint(m, y .<= totalInvestment * x)
@constraint(m, sum{y[i] , i=1:n} <= totalInvestment)

for i=1:n
    @constraint(m, minInvestment[i] * x[i] <= y[i])
    @constraint(m, y[i] * x[i] <= maxInvestment[i])
end

# cannot invest more than the sum of 2, 4, and 6 into 5. 
@constraint(m, y[5] <= y[2] + y[4] + y[6])

# ensure that the amount invsted into 3 is the same as 6; i.e. 1 invest to 3, we must 1 invest to 6
@constraint(m, x[3] <= x[6])

@objective(m, Max, sum{y[i] * expectedRtn[i], i=1:n})

status = solve(m)
println(status)
println("x= ", getvalue(x))
println("y= ", getvalue(y))
println("objective = ", getobjectivevalue(m))

Academic license - for non-commercial use only
Optimal
x= [0.0, 0.0, 1.0, 1.0, 1.0, 1.0]
y= [0.0, 0.0, 35.0, 5.0, 22.5, 17.5]
objective = 13.5


For the highest return profit, we should look to invest 35 million into Option 3, 5 million into option 4, 22.5 million into option 5, and 17.5 million into option 6.